### PR TITLE
some docstrings

### LINKED ISSUE #275: `__init__` needs a docstring
So that when users type `ipywidgets?`, at least *something* is printed...

We should *never* have top-level modules with zero documentation, never.

### PR DESCRIPTION
**Pull request by minrk**
Add docstring to top-level ipywidgets module and the integer widgets.

I think there are a *lot* more docstrings we need to add/improve, but this is a start.

closes #275

### CODE REVIEW

**Comment by jdfreder** [comment]
Thanks min!

File: ipywidgets/__init__.py [hunk 1]
```diff
@@ -1,3 +1,21 @@
+"""IPython widgets for Jupyter notebooks.
+
+IPython Widgets provide simple interactive controls in the notebook.
+Widgets provide a mechanism for creating intera
```
Comment (jdfreder) [inline ipywidgets/__init__.py]: Whoops, forgot to delete this line?

File: ipywidgets/__init__.py [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+Each Widget corresponds to an object in Python and Javascript,
+with controls on the page.
+
+To put a Widget on the page, you can display it with IPython's display machinery::
```
Comment (jdfreder) [inline ipywidgets/__init__.py]: Double ::

Comment (jdfreder) [inline ipywidgets/__init__.py]: Ah, rst

**Comment by jdfreder** [comment]
Sent a PR - https://github.com/minrk/ipywidgets/pull/2

**Comment by minrk** [comment]
Thanks, @jdfreder!

**Comment by fperez** [comment]
Thanks so much, folks!!

**Pull request merged**